android-components/components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/manager/FxaAccountManager.kt

```diff
@@ -551,14 +558,32 @@ open class FxaAccountManager(
         via: Event,
     ): Event? = when (forState.progressState) {
         ProgressState.Initializing -> {
+            appServicesStateMachineChecker.checkInternalState(FxaStateCheckerState.GetAuthState)
```

Comment:
It looks like all the branches take in consideration current state, via event and the result progress state and call `AppServicesStateMachineChecker` respectively. If we refactor the checker to have something like

```
internal fun validateProgressState(
        forState: State.Active,
        via: Event,
        result: Event,
    ) {
        when (forState.progressState) {
            ProgressState.Initializing -> {
                rustChecker.checkInternalState(FxaStateCheckerState.GetAuthState)
                when (result) {
                    Event.Progress.AccountNotFound -> {
                        rustChecker.handleInternalEvent(
                            FxaStateCheckerEvent.GetAuthStateSuccess(FxaRustAuthState.DISCONNECTED),
                        )
                    }
                    Event.Progress.AccountNotFound -> {
                        rustChecker.handleInternalEvent(
                            FxaStateCheckerEvent.GetAuthStateSuccess(FxaRustAuthState.CONNECTED),
                        )
                    }
                    else -> {}
                }
            }

            else -> {}
        }
    }

we could move the branching logic there and reduce the change to the `FxaAccountManager.kt` itself.

The exceptions are `Event.Account.BeginEmailFlow` and `Event.Account.BeginPairingFlow`. 

I wonder if we could make it more neat? I would move out of `FxaAccountManager` class as much code as possible. Wonder what you think.

Reply:
This makes sense to me.  I moved all the logic I could into that class and the only exceptions left were the complete flow events, there's much less changes to `FxaAccountManager` now.